messaging/src/main/java/org/axonframework/messaging/MessageHandlerTimeout.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.messaging;
+
+import org.axonframework.messaging.annotation.HasHandlerAttributes;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Sets the timeout settings of a message handler. Takes precedence over configuration through
+ * {@link org.axonframework.messaging.timeout.HandlerTimeoutConfiguration}, for each property individually.
+ * <p>
+ * Any property set to {@code -1} will use the default value of the
+ * {@link org.axonframework.messaging.timeout.HandlerTimeoutConfiguration}. For example, setting the timeout to
+ * {@code -1}, and {@code warningThreshold} to {@code 1000}, will take the default timeout from the configuration, while
+ * setting the warning threshold to {@code 1000}.
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.11
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.METHOD, ElementType.CONSTRUCTOR})
+@HasHandlerAttributes
+public @interface MessageHandlerTimeout {
+
+    /**
+     * The timeout for a single message in milliseconds. A value of {@code -1} indicates that the timeout should be used
+     * of the {@link org.axonframework.messaging.timeout.HandlerTimeoutConfiguration}.
+     *
+     * @return the timeout for a single message handler in milliseconds
+     */
+    int timeout() default -1;
```

Comment:
Wdyt about adding `ms` to all the fields here?

Reply:
I considered it. But, for example, the Spriong Transactional annotation doesn't do this either:

```
@Transactional(timeout = 30)
```
![image](https://github.com/user-attachments/assets/8e03ee6f-dda3-4ce2-81da-c2a633f154e8)


Although I now realize, Spring does it in seconds. I will add ms here as well ;)